server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -172,7 +177,19 @@ public Stream<Reference> getReferences() {
 
   @Override
   public Set<TableId> getTableIDs() {
-    return context.getTableIdToNameMap().keySet();
+    final String tablesPath = context.getZooKeeperRoot() + Constants.ZTABLES;
+    final ZooReader zr = context.getZooReader();
+    final Set<TableId> tids = new HashSet<>();
+    while (true) {
+      try {
+        zr.sync(tablesPath);
+        zr.getChildren(tablesPath).forEach(t -> tids.add(TableId.of(t)));
+        return tids;
+      } catch (KeeperException | InterruptedException e) {
+        log.error("Error getting tables from ZooKeeper, retrying", e);
+        UtilWaitThread.sleepUninterruptibly(1, TimeUnit.SECONDS);
+      }
```

Comment:
First, this shouldn't loop if it's explicitly interrupted. The process should probably die in that case. Second, if this does loop because of a KeeperException, the set of table IDs that was constructed prior to the `while (true)` loop could include deleted tables. Is that okay/intended? If so, a small inline comment explaining why that's okay might be useful for future onlookers.

Reply:
Addressed in 1f8d37305789fa1540158e32b54da86a7af3ae87